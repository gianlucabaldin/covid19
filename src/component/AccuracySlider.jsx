import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Slider from '@material-ui/core/Slider';
import { DEFAUL_MAX_DATA_SIZE } from '../utils/consts';

const useStyles = makeStyles((theme) => ({
  root: {
    width: 200,
    marginRight: 25,
    textAlign: 'center',
  },
  margin: {
    height: theme.spacing(3),
  },
}));

// the marks used as base
const marks = [
  {
    value: 10,
    label: '10',
  },
  {
    value: 20,
    label: '20',
  },
  {
    value: 30,
    label: '30',
  },
  {
    value: 40,
    label: '40',
  },
  {
    value: 50,
    label: '50',
  },
];

function valuetext(value) {
  return `${value}Â°C`;
}

export default function AccuracySlider({
  accuracyValue,
  disabled,
  changeAccuracy,
}) {
  const classes = useStyles();

  /**
   * Called when its value has been changed, calls the parent method
   * with the new value
   * @param {text} e the event object contains the new value
   */
  const onChangeAccuracy = (e) =>
    changeAccuracy(
      e && e.target && e.target.ariaValueNow
        ? parseInt(e.target.ariaValueNow, 10)
        : 0,
    );

  return (
    <div className={classes.root}>
      <Typography
        id="discrete-slider-custom"
        textAlign="center"
        style={disabled ? { color: 'grey.500' } : {}}
      >
        Accuracy
      </Typography>
      <Slider
        disabled={disabled}
        defaultValue={accuracyValue || DEFAUL_MAX_DATA_SIZE}
        getAriaValueText={valuetext}
        aria-labelledby="discrete-slider-custom"
        step={10}
        marks={marks}
        min={10}
        max={50}
        onChangeCommitted={onChangeAccuracy}
      />
    </div>
  );
}
