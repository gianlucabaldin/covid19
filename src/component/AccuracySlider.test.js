import React from 'react';
import { shallow } from 'enzyme';
import AccuracySlider from './AccuracySlider';

describe('AccuracySlider', () => {
  const wrapper = shallow(<AccuracySlider />);
  it('should renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
