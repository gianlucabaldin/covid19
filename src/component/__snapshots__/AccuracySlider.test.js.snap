// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccuracySlider should renders properly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    id="discrete-slider-custom"
    style={Object {}}
    textAlign="center"
  >
    Accuracy
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Slider))
    aria-labelledby="discrete-slider-custom"
    getAriaValueText={[Function]}
    marks={
      Array [
        Object {
          "label": "10",
          "value": 10,
        },
        Object {
          "label": "20",
          "value": 20,
        },
        Object {
          "label": "30",
          "value": 30,
        },
        Object {
          "label": "40",
          "value": 40,
        },
        Object {
          "label": "50",
          "value": 50,
        },
      ]
    }
    max={50}
    min={10}
    onChangeCommitted={[Function]}
    step={10}
  />
</div>
`;
